apps/meteor/ee/app/livechat-enterprise/server/hooks/beforeNewInquiry.js

```diff
@@ -13,24 +13,38 @@ callbacks.add(
 			return extraData;
 		}
 
-		const sla = Promise.await(
-			OmnichannelServiceLevelAgreements.findOneByIdOrName(searchTerm, {
-				projection: { dueTimeInMinutes: 1 },
-			}),
-		);
-		if (!sla) {
-			throw new Meteor.Error('error-invalid-priority', 'Invalid sla', {
+		const slaP = OmnichannelServiceLevelAgreements.findOneByIdOrName(searchTerm, {
+			projection: { dueTimeInMinutes: 1 },
+		});
+		const priorityP = LivechatPriority.findOneByIdOrName(searchTerm, {
+			projection: { _id: 1, priorityWeight: 1 },
+		});
+		const [sla, priority] = Promise.await([slaP, priorityP]);
+		if (!sla && !priority) {
+			throw new Meteor.Error('error-invalid-priority', 'Invalid sla or priority', {
+				function: 'livechat.beforeInquiry',
+			});
+		}
+		if (sla && priority) {
+			throw new Meteor.Error('error-invalid-state', 'You cannot have both sla and priority assigned to the same inquiry at a time', {
 				function: 'livechat.beforeInquiry',
 			});
 		}
-
 		const now = new Date();
 		const ts = new Date(now.getTime());
-		const { dueTimeInMinutes: estimatedWaitingTimeQueue } = sla;
-		const estimatedServiceTimeAt = new Date(now.setMinutes(now.getMinutes() + estimatedWaitingTimeQueue));
-
+		const changes = {
+			ts,
+		};
+		if (sla) {
+			changes.estimatedWaitingTimeQueue = sla.dueTimeInMinutes;
+			changes.estimatedServiceTimeAt = new Date(now.setMinutes(now.getMinutes() + sla.dueTimeInMinutes));
+		}
+		if (priority) {
+			changes.priorityId = priority._id;
+			changes.priorityWeight = priority.priorityWeight;
+		}
 		cbLogger.debug('Callback success. Queue timing properties added');
-		return Object.assign({ ...props }, { ts, estimatedWaitingTimeQueue, estimatedServiceTimeAt });
+		return Object.assign({ ...props }, { ...changes });
```

Comment:
Any advantages on using `Object.assign` + spread, instead of just spread? (Ex.: `{ ...props, ...changes }`)

Reply:
Nice catch! Don't think Object.assign is needed here. Removed it now :heavy_check_mark: